# FEATURE SELECTION EDASPY

# INFO

El objetivo de este _notebook_ es la realización de _feature selection_ mediante algoritmos evolutivos.

## LIBRARIES

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [2]:
import numpy as np
import pandas as pd
import matplotlib
from matplotlib import pyplot as plt # plots
import seaborn as sns # plots
from scipy import stats as sts # normality test JB
from sklearn import model_selection
from sklearn import linear_model
from sklearn import metrics
from sklearn.linear_model import Ridge, RidgeCV # Ridge
from sklearn.linear_model import Lasso, LassoCV # Lasso
from sklearn.linear_model import LogisticRegression # Log Regression
from sklearn.metrics import r2_score # R^2
from sklearn.metrics import mean_squared_error # MSE
from sklearn import preprocessing # estandarization
from sklearn.ensemble import IsolationForest # outliers
from math import sqrt # sqrt
import itertools # aplanar arrays
import math
from sklearn.svm import SVC # SVM
from sklearn.metrics import classification_report, confusion_matrix
from sklearn.ensemble import RandomForestRegressor, RandomForestClassifier
import xgboost as xgb # XGBOOST
from sklearn.metrics import roc_curve, roc_auc_score, plot_roc_curve # ROC Curve
from sklearn.datasets import make_classification
import random
from sklearn.model_selection import train_test_split # split data
import pickle
import lightgbm as lgb
from EDAspy.optimization.univariate import EDA_discrete as EDAd
from sklearn.metrics import f1_score

C:\Users\migue\Anaconda3\lib\site-packages\statsmodels\tools\_testing.py:19: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.
  import pandas.util.testing as tm


## READING DATA

Carga y lectura de los datos con las transformaciones ya realizadas adecuadas a un modelo Lightgbm.

In [3]:
data_path = "../data/datos_procesados.csv"

In [4]:
datos = pd.read_csv(data_path)

In [5]:
datos

TARGET  CODE_GENDER  FLAG_OWN_CAR  FLAG_OWN_REALTY  CNT_CHILDREN  \
0            1            0             0                0             0   
1            0            1             0                1             0   
2            0            0             1                0             0   
3            0            1             0                0             0   
4            0            0             0                0             0   
...        ...          ...           ...              ...           ...   
307502       0            0             0                1             0   
307503       0            1             0                0             0   
307504       0            1             0                0             0   
307505       1            1             0                0             0   
307506       0            1             0                1             0   

        AMT_INCOME_TOTAL  AMT_CREDIT  AMT_ANNUITY  AMT_GOODS_PRICE  \
0               202500.0    406597.5      24700.5         351000.0   
1               270000.0   1293502.5      35698.5        1129500.0   
2                67500.0    135000.0       6750.0         135000.0   
3               135000.0    312682.5      29686.5         297000.0   
4               121500.0    513000.0      21865.5         513000.0   
...                  ...         ...          ...              ...   
307502          157500.0    254700.0      27558.0         225000.0   
307503           72000.0    269550.0      12001.5         225000.0   
307504          153000.0    677664.0      29979.0         585000.0   
307505          171000.0    370107.0      20205.0         319500.0   
307506          157500.0    675000.0      49117.5         675000.0   

        REGION_POPULATION_RELATIVE  ...  CC_NAME_CONTRACT_STATUS_Signed_MAX  \
0                         0.018801  ...                                 NaN   
1                         0.003541  ...                                 NaN   
2                         0.010032  ...                                 NaN   
3                         0.008019  ...                                 0.0   
4                         0.028663  ...                                 NaN   
...                            ...  ...                                 ...   
307502                    0.032561  ...                                 NaN   
307503                    0.025164  ...                                 NaN   
307504                    0.005002  ...                                 NaN   
307505                    0.005313  ...                                 NaN   
307506                    0.046220  ...                                 NaN   

        CC_NAME_CONTRACT_STATUS_Signed_MEAN  \
0                                       NaN   
1                                       NaN   
2                                       NaN   
3                                       0.0   
4                                       NaN   
...                                     ...   
307502                                  NaN   
307503                                  NaN   
307504                                  NaN   
307505                                  NaN   
307506                                  NaN   

        CC_NAME_CONTRACT_STATUS_Signed_SUM  \
0                                      NaN   
1                                      NaN   
2                                      NaN   
3                                      0.0   
4                                      NaN   
...                                    ...   
307502                                 NaN   
307503                                 NaN   
307504                                 NaN   
307505                                 NaN   
307506                                 NaN   

        CC_NAME_CONTRACT_STATUS_Signed_VAR  CC_NAME_CONTRACT_STATUS_nan_MIN  \
0                                      NaN                              NaN   
1                            

In [6]:
datos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 307511 entries, 0 to 307510
Columns: 122 entries, SK_ID_CURR to AMT_REQ_CREDIT_BUREAU_YEAR
dtypes: float64(65), int64(41), object(16)
memory usage: 286.2+ MB


We have:
* 307 511 entries.
* 122 variables:
    * 65 float.
    * 41 int.
    * 16 object, strings.

División en Train y Test

In [18]:
X = datos.loc[:, datos.columns != 'TARGET']
Y = datos['TARGET']

In [19]:
X_train, X_test, y_train, y_test = train_test_split(X, Y, test_size=0.20, random_state=32)

Se incluyen los hiperparámetros del modelo Lightgbm que obtuvo buenos resultados en la competición de Kaggle. La idea de comenzar con estos hiperparámetros es que se sabe que funcionan de forma adecuada y, por tanto, con los cambios que se han incluído, como por ejemplo asignar pesos distintos a cada una de las clases del TARGET, se podrán encontrar unos mejores hiperparámetros de forma más rápida.

In [21]:
modelo_final = lgb.LGBMClassifier(
            n_estimators=3000,
            learning_rate=0.02,
            num_leaves=34,
            colsample_bytree=0.9497036,
            subsample=0.8715623,
            max_depth=8,
            reg_alpha=0.041545473,
            reg_lambda=0.0735294,
            min_split_gain=0.0222415,
            min_child_weight=39.3259775,
            silent=-1,
            class_weight={0:1, 1:4.7})

Definición de la función de coste que se quiere maximizar con el uso de los algoritmos evolutivos EDAs.

In [22]:
def f1_gabri(y_test, y_pred):
    metric = (f1_score(y_test, y_pred, pos_label=0) * f1_score(y_test, y_pred, pos_label=1))**(1/2)
    return metric

In [24]:
def lgb_f1_score(y_pred, y_test):
    y_test = np.round(y_test) # scikits f1 doesn't like probabilities
    return 'f1_gabri', (f1_score(y_test, y_pred, pos_label=0) * f1_score(y_test, y_pred, pos_label=1))**(1/2), True

In [25]:
def costVarF1GeoF():
    def costVarF1Geo(dict_var):
        """
        - modelRB hace referencia al modelo Lightgbm que ha sido asignado anteriormente
        - para cada selección de diversas variables se hace un fit sobre la parte de train
        - se realiza una predicción para cada entrenamiento del modelo
        - se calcula y evalúa de la misma forma la función de coste que se busca maximizar
        
        """
        
        modelRB = modelo_final
        modelRB_feature_selection = [variable for variable in dict_var.keys() if dict_var[variable] == 1]
        modelRB.fit(X_train[modelRB_feature_selection], y_train,
                    eval_set=[(X_train[modelRB_feature_selection], y_train),
                              (X_test[modelRB_feature_selection], y_test)],
                    eval_metric= lgb_f1_score, verbose= 100, early_stopping_rounds= 150)
        resultsRB = modelRB.predict(X_test[modelRB_feature_selection])
        #f1g = f1_score(y_test, resultsRB)
        f1g = (f1_score(y_test, resultsRB, pos_label=0) * f1_score(y_test, resultsRB, pos_label=1))**(1/2)
        print(f1g)
        return f1g
    return costVarF1Geo

In [26]:
vector = pd.DataFrame(columns=X_train.columns.values)
vector.loc[0] = 0.5
costVarF1GeouEDF = costVarF1GeoF()
EDAuEDF = EDAd(SIZE_GEN=20, MAX_IT=10, DEAD_ITER=5, ALPHA=0.35, vector=vector,
            aim='maximize', cost_function=costVarF1GeouEDF)

- bestcost: muestra el valor más alto que se ha conseguido para la función de coste
- solution: refleja las variables seleccionadas con las que se ha conseguido el mayor valor para el f1 score geométrico
- history: muestra la evolución del valor de la función de coste que se buscaba maximizar a lo largo de las iteraciones

In [27]:
bestcost, solution, history = EDAuEDF.run(output=True)

Training until validation scores don't improve for 150 rounds
[100]	valid_0's binary_logloss: 0.369195	valid_0's f1_gabri: 0.486542	valid_1's binary_logloss: 0.372165	valid_1's f1_gabri: 0.46427
[200]	valid_0's binary_logloss: 0.35434	valid_0's f1_gabri: 0.547469	valid_1's binary_logloss: 0.360021	valid_1's f1_gabri: 0.519949
[300]	valid_0's binary_logloss: 0.345938	valid_0's f1_gabri: 0.566111	valid_1's binary_logloss: 0.354403	valid_1's f1_gabri: 0.531218
[400]	valid_0's binary_logloss: 0.339824	valid_0's f1_gabri: 0.578812	valid_1's binary_logloss: 0.350895	valid_1's f1_gabri: 0.536995
[500]	valid_0's binary_logloss: 0.334542	valid_0's f1_gabri: 0.588027	valid_1's binary_logloss: 0.348144	valid_1's f1_gabri: 0.540024
[600]	valid_0's binary_logloss: 0.329893	valid_0's f1_gabri: 0.596624	valid_1's binary_logloss: 0.345844	valid_1's f1_gabri: 0.542431
[700]	valid_0's binary_logloss: 0.325832	valid_0's f1_gabri: 0.604708	valid_1's binary_logloss: 0.34407	valid_1's f1_gabri: 0.544156
[80

[900]	valid_0's binary_logloss: 0.321618	valid_0's f1_gabri: 0.616689	valid_1's binary_logloss: 0.344438	valid_1's f1_gabri: 0.5327
Early stopping, best iteration is:
[791]	valid_0's binary_logloss: 0.325776	valid_0's f1_gabri: 0.608242	valid_1's binary_logloss: 0.346136	valid_1's f1_gabri: 0.533679
0.533678588272024
Training until validation scores don't improve for 150 rounds
[100]	valid_0's binary_logloss: 0.36365	valid_0's f1_gabri: 0.517178	valid_1's binary_logloss: 0.3662	valid_1's f1_gabri: 0.495299
[200]	valid_0's binary_logloss: 0.348621	valid_0's f1_gabri: 0.562384	valid_1's binary_logloss: 0.353929	valid_1's f1_gabri: 0.533134
[300]	valid_0's binary_logloss: 0.340063	valid_0's f1_gabri: 0.57918	valid_1's binary_logloss: 0.348083	valid_1's f1_gabri: 0.546133
[400]	valid_0's binary_logloss: 0.33377	valid_0's f1_gabri: 0.588944	valid_1's binary_logloss: 0.344508	valid_1's f1_gabri: 0.551706
[500]	valid_0's binary_logloss: 0.32857	valid_0's f1_gabri: 0.598515	valid_1's binary_lo

Early stopping, best iteration is:
[661]	valid_0's binary_logloss: 0.331348	valid_0's f1_gabri: 0.597988	valid_1's binary_logloss: 0.347779	valid_1's f1_gabri: 0.534411
0.534410627168919
Training until validation scores don't improve for 150 rounds
[100]	valid_0's binary_logloss: 0.371743	valid_0's f1_gabri: 0.472792	valid_1's binary_logloss: 0.374302	valid_1's f1_gabri: 0.453709
[200]	valid_0's binary_logloss: 0.356648	valid_0's f1_gabri: 0.537384	valid_1's binary_logloss: 0.361871	valid_1's f1_gabri: 0.511668
[300]	valid_0's binary_logloss: 0.347942	valid_0's f1_gabri: 0.558155	valid_1's binary_logloss: 0.355773	valid_1's f1_gabri: 0.528739
[400]	valid_0's binary_logloss: 0.341723	valid_0's f1_gabri: 0.571846	valid_1's binary_logloss: 0.35215	valid_1's f1_gabri: 0.535713
[500]	valid_0's binary_logloss: 0.336605	valid_0's f1_gabri: 0.583287	valid_1's binary_logloss: 0.349471	valid_1's f1_gabri: 0.540492
[600]	valid_0's binary_logloss: 0.332091	valid_0's f1_gabri: 0.592702	valid_1's bi

0.555414038982567
Training until validation scores don't improve for 150 rounds
[100]	valid_0's binary_logloss: 0.371277	valid_0's f1_gabri: 0.474542	valid_1's binary_logloss: 0.373388	valid_1's f1_gabri: 0.454169
[200]	valid_0's binary_logloss: 0.357174	valid_0's f1_gabri: 0.539526	valid_1's binary_logloss: 0.362059	valid_1's f1_gabri: 0.516606
[300]	valid_0's binary_logloss: 0.348744	valid_0's f1_gabri: 0.559613	valid_1's binary_logloss: 0.356448	valid_1's f1_gabri: 0.528097
[400]	valid_0's binary_logloss: 0.342343	valid_0's f1_gabri: 0.573826	valid_1's binary_logloss: 0.352625	valid_1's f1_gabri: 0.535478
[500]	valid_0's binary_logloss: 0.337029	valid_0's f1_gabri: 0.584705	valid_1's binary_logloss: 0.349802	valid_1's f1_gabri: 0.537073
[600]	valid_0's binary_logloss: 0.332442	valid_0's f1_gabri: 0.594522	valid_1's binary_logloss: 0.34765	valid_1's f1_gabri: 0.53983
[700]	valid_0's binary_logloss: 0.328253	valid_0's f1_gabri: 0.602124	valid_1's binary_logloss: 0.345767	valid_1's f1_

Training until validation scores don't improve for 150 rounds
[100]	valid_0's binary_logloss: 0.369586	valid_0's f1_gabri: 0.482995	valid_1's binary_logloss: 0.37232	valid_1's f1_gabri: 0.456611
[200]	valid_0's binary_logloss: 0.3538	valid_0's f1_gabri: 0.54706	valid_1's binary_logloss: 0.358919	valid_1's f1_gabri: 0.515029
[300]	valid_0's binary_logloss: 0.344816	valid_0's f1_gabri: 0.567251	valid_1's binary_logloss: 0.352542	valid_1's f1_gabri: 0.535403
[400]	valid_0's binary_logloss: 0.338401	valid_0's f1_gabri: 0.579469	valid_1's binary_logloss: 0.348614	valid_1's f1_gabri: 0.539542
[500]	valid_0's binary_logloss: 0.333288	valid_0's f1_gabri: 0.590884	valid_1's binary_logloss: 0.345861	valid_1's f1_gabri: 0.544197
[600]	valid_0's binary_logloss: 0.328862	valid_0's f1_gabri: 0.60002	valid_1's binary_logloss: 0.343806	valid_1's f1_gabri: 0.546027
[700]	valid_0's binary_logloss: 0.32484	valid_0's f1_gabri: 0.607562	valid_1's binary_logloss: 0.342129	valid_1's f1_gabri: 0.547213
[800]	

[500]	valid_0's binary_logloss: 0.331362	valid_0's f1_gabri: 0.594226	valid_1's binary_logloss: 0.344111	valid_1's f1_gabri: 0.550611
[600]	valid_0's binary_logloss: 0.326604	valid_0's f1_gabri: 0.603093	valid_1's binary_logloss: 0.341883	valid_1's f1_gabri: 0.553743
[700]	valid_0's binary_logloss: 0.3224	valid_0's f1_gabri: 0.611686	valid_1's binary_logloss: 0.340108	valid_1's f1_gabri: 0.555034
[800]	valid_0's binary_logloss: 0.318439	valid_0's f1_gabri: 0.619753	valid_1's binary_logloss: 0.338386	valid_1's f1_gabri: 0.555954
[900]	valid_0's binary_logloss: 0.314706	valid_0's f1_gabri: 0.626909	valid_1's binary_logloss: 0.336837	valid_1's f1_gabri: 0.5556
[1000]	valid_0's binary_logloss: 0.311047	valid_0's f1_gabri: 0.634347	valid_1's binary_logloss: 0.335356	valid_1's f1_gabri: 0.555143
Early stopping, best iteration is:
[877]	valid_0's binary_logloss: 0.315497	valid_0's f1_gabri: 0.625887	valid_1's binary_logloss: 0.337131	valid_1's f1_gabri: 0.557084
0.5570839578096559
Training un

[500]	valid_0's binary_logloss: 0.336575	valid_0's f1_gabri: 0.58446	valid_1's binary_logloss: 0.349481	valid_1's f1_gabri: 0.541033
[600]	valid_0's binary_logloss: 0.331997	valid_0's f1_gabri: 0.592776	valid_1's binary_logloss: 0.347363	valid_1's f1_gabri: 0.541184
[700]	valid_0's binary_logloss: 0.327875	valid_0's f1_gabri: 0.601852	valid_1's binary_logloss: 0.345585	valid_1's f1_gabri: 0.543474
[800]	valid_0's binary_logloss: 0.32392	valid_0's f1_gabri: 0.610084	valid_1's binary_logloss: 0.343934	valid_1's f1_gabri: 0.542831
[900]	valid_0's binary_logloss: 0.32013	valid_0's f1_gabri: 0.618672	valid_1's binary_logloss: 0.342416	valid_1's f1_gabri: 0.543323
[1000]	valid_0's binary_logloss: 0.31641	valid_0's f1_gabri: 0.626716	valid_1's binary_logloss: 0.34099	valid_1's f1_gabri: 0.543354
Early stopping, best iteration is:
[918]	valid_0's binary_logloss: 0.319457	valid_0's f1_gabri: 0.619908	valid_1's binary_logloss: 0.342167	valid_1's f1_gabri: 0.544407
0.5444065029996368
Training unt

[200]	valid_0's binary_logloss: 0.353848	valid_0's f1_gabri: 0.548427	valid_1's binary_logloss: 0.35911	valid_1's f1_gabri: 0.524209
[300]	valid_0's binary_logloss: 0.344853	valid_0's f1_gabri: 0.569868	valid_1's binary_logloss: 0.352778	valid_1's f1_gabri: 0.537478
[400]	valid_0's binary_logloss: 0.338242	valid_0's f1_gabri: 0.582355	valid_1's binary_logloss: 0.348719	valid_1's f1_gabri: 0.546342
[500]	valid_0's binary_logloss: 0.332827	valid_0's f1_gabri: 0.591447	valid_1's binary_logloss: 0.345696	valid_1's f1_gabri: 0.550984
[600]	valid_0's binary_logloss: 0.328121	valid_0's f1_gabri: 0.601498	valid_1's binary_logloss: 0.343364	valid_1's f1_gabri: 0.554552
[700]	valid_0's binary_logloss: 0.323825	valid_0's f1_gabri: 0.610297	valid_1's binary_logloss: 0.341382	valid_1's f1_gabri: 0.555844
[800]	valid_0's binary_logloss: 0.319947	valid_0's f1_gabri: 0.618424	valid_1's binary_logloss: 0.339709	valid_1's f1_gabri: 0.55723
[900]	valid_0's binary_logloss: 0.316197	valid_0's f1_gabri: 0.6

Training until validation scores don't improve for 150 rounds
[100]	valid_0's binary_logloss: 0.363607	valid_0's f1_gabri: 0.518631	valid_1's binary_logloss: 0.36606	valid_1's f1_gabri: 0.493211
[200]	valid_0's binary_logloss: 0.347905	valid_0's f1_gabri: 0.564706	valid_1's binary_logloss: 0.352797	valid_1's f1_gabri: 0.537035
[300]	valid_0's binary_logloss: 0.339144	valid_0's f1_gabri: 0.582425	valid_1's binary_logloss: 0.346696	valid_1's f1_gabri: 0.551341
[400]	valid_0's binary_logloss: 0.332724	valid_0's f1_gabri: 0.592622	valid_1's binary_logloss: 0.342926	valid_1's f1_gabri: 0.557355
[500]	valid_0's binary_logloss: 0.327404	valid_0's f1_gabri: 0.600975	valid_1's binary_logloss: 0.340109	valid_1's f1_gabri: 0.559927
[600]	valid_0's binary_logloss: 0.322869	valid_0's f1_gabri: 0.608444	valid_1's binary_logloss: 0.337991	valid_1's f1_gabri: 0.562185
[700]	valid_0's binary_logloss: 0.318634	valid_0's f1_gabri: 0.616465	valid_1's binary_logloss: 0.336144	valid_1's f1_gabri: 0.562859
[

[200]	valid_0's binary_logloss: 0.348676	valid_0's f1_gabri: 0.562803	valid_1's binary_logloss: 0.353513	valid_1's f1_gabri: 0.536792
[300]	valid_0's binary_logloss: 0.339933	valid_0's f1_gabri: 0.57962	valid_1's binary_logloss: 0.347447	valid_1's f1_gabri: 0.549421
[400]	valid_0's binary_logloss: 0.333497	valid_0's f1_gabri: 0.590497	valid_1's binary_logloss: 0.343592	valid_1's f1_gabri: 0.55665
[500]	valid_0's binary_logloss: 0.32825	valid_0's f1_gabri: 0.600531	valid_1's binary_logloss: 0.340878	valid_1's f1_gabri: 0.559866
[600]	valid_0's binary_logloss: 0.323668	valid_0's f1_gabri: 0.607801	valid_1's binary_logloss: 0.338856	valid_1's f1_gabri: 0.561259
[700]	valid_0's binary_logloss: 0.319541	valid_0's f1_gabri: 0.615221	valid_1's binary_logloss: 0.337115	valid_1's f1_gabri: 0.562309
[800]	valid_0's binary_logloss: 0.315594	valid_0's f1_gabri: 0.623254	valid_1's binary_logloss: 0.335477	valid_1's f1_gabri: 0.562878
[900]	valid_0's binary_logloss: 0.311939	valid_0's f1_gabri: 0.63

[700]	valid_0's binary_logloss: 0.325531	valid_0's f1_gabri: 0.604838	valid_1's binary_logloss: 0.343097	valid_1's f1_gabri: 0.548593
[800]	valid_0's binary_logloss: 0.321605	valid_0's f1_gabri: 0.611287	valid_1's binary_logloss: 0.341395	valid_1's f1_gabri: 0.548777
[900]	valid_0's binary_logloss: 0.317898	valid_0's f1_gabri: 0.619851	valid_1's binary_logloss: 0.339839	valid_1's f1_gabri: 0.549803
[1000]	valid_0's binary_logloss: 0.314214	valid_0's f1_gabri: 0.627333	valid_1's binary_logloss: 0.338272	valid_1's f1_gabri: 0.549159
Early stopping, best iteration is:
[853]	valid_0's binary_logloss: 0.319607	valid_0's f1_gabri: 0.615568	valid_1's binary_logloss: 0.340557	valid_1's f1_gabri: 0.55045
0.5504504297255778
Training until validation scores don't improve for 150 rounds
[100]	valid_0's binary_logloss: 0.364933	valid_0's f1_gabri: 0.514526	valid_1's binary_logloss: 0.367379	valid_1's f1_gabri: 0.491775
[200]	valid_0's binary_logloss: 0.350571	valid_0's f1_gabri: 0.557066	valid_1's 

[300]	valid_0's binary_logloss: 0.343919	valid_0's f1_gabri: 0.570319	valid_1's binary_logloss: 0.351714	valid_1's f1_gabri: 0.534707
[400]	valid_0's binary_logloss: 0.337327	valid_0's f1_gabri: 0.583373	valid_1's binary_logloss: 0.347567	valid_1's f1_gabri: 0.541902
[500]	valid_0's binary_logloss: 0.331896	valid_0's f1_gabri: 0.594439	valid_1's binary_logloss: 0.344582	valid_1's f1_gabri: 0.548212
[600]	valid_0's binary_logloss: 0.327299	valid_0's f1_gabri: 0.603987	valid_1's binary_logloss: 0.342308	valid_1's f1_gabri: 0.549844
[700]	valid_0's binary_logloss: 0.323129	valid_0's f1_gabri: 0.612039	valid_1's binary_logloss: 0.34046	valid_1's f1_gabri: 0.552153
[800]	valid_0's binary_logloss: 0.319268	valid_0's f1_gabri: 0.619671	valid_1's binary_logloss: 0.338756	valid_1's f1_gabri: 0.550684
[900]	valid_0's binary_logloss: 0.315608	valid_0's f1_gabri: 0.628104	valid_1's binary_logloss: 0.337176	valid_1's f1_gabri: 0.553973
[1000]	valid_0's binary_logloss: 0.311979	valid_0's f1_gabri: 0

Training until validation scores don't improve for 150 rounds
[100]	valid_0's binary_logloss: 0.363986	valid_0's f1_gabri: 0.518561	valid_1's binary_logloss: 0.366412	valid_1's f1_gabri: 0.493001
[200]	valid_0's binary_logloss: 0.349354	valid_0's f1_gabri: 0.561096	valid_1's binary_logloss: 0.354345	valid_1's f1_gabri: 0.535593
[300]	valid_0's binary_logloss: 0.341148	valid_0's f1_gabri: 0.57684	valid_1's binary_logloss: 0.348739	valid_1's f1_gabri: 0.546508
[400]	valid_0's binary_logloss: 0.334905	valid_0's f1_gabri: 0.589876	valid_1's binary_logloss: 0.345034	valid_1's f1_gabri: 0.55237
[500]	valid_0's binary_logloss: 0.329684	valid_0's f1_gabri: 0.598287	valid_1's binary_logloss: 0.342288	valid_1's f1_gabri: 0.554909
[600]	valid_0's binary_logloss: 0.325169	valid_0's f1_gabri: 0.605905	valid_1's binary_logloss: 0.340252	valid_1's f1_gabri: 0.556521
[700]	valid_0's binary_logloss: 0.320998	valid_0's f1_gabri: 0.614184	valid_1's binary_logloss: 0.3385	valid_1's f1_gabri: 0.557136
[800

[300]	valid_0's binary_logloss: 0.339973	valid_0's f1_gabri: 0.579728	valid_1's binary_logloss: 0.347474	valid_1's f1_gabri: 0.547223
[400]	valid_0's binary_logloss: 0.333482	valid_0's f1_gabri: 0.590358	valid_1's binary_logloss: 0.34353	valid_1's f1_gabri: 0.554261
[500]	valid_0's binary_logloss: 0.328042	valid_0's f1_gabri: 0.600437	valid_1's binary_logloss: 0.340605	valid_1's f1_gabri: 0.55849
[600]	valid_0's binary_logloss: 0.323427	valid_0's f1_gabri: 0.608308	valid_1's binary_logloss: 0.338352	valid_1's f1_gabri: 0.56036
[700]	valid_0's binary_logloss: 0.319235	valid_0's f1_gabri: 0.616166	valid_1's binary_logloss: 0.336511	valid_1's f1_gabri: 0.561065
[800]	valid_0's binary_logloss: 0.315243	valid_0's f1_gabri: 0.622445	valid_1's binary_logloss: 0.334841	valid_1's f1_gabri: 0.560994
[900]	valid_0's binary_logloss: 0.311461	valid_0's f1_gabri: 0.629541	valid_1's binary_logloss: 0.33332	valid_1's f1_gabri: 0.562763
[1000]	valid_0's binary_logloss: 0.307776	valid_0's f1_gabri: 0.63

[300]	valid_0's binary_logloss: 0.340254	valid_0's f1_gabri: 0.578944	valid_1's binary_logloss: 0.347624	valid_1's f1_gabri: 0.548124
[400]	valid_0's binary_logloss: 0.333952	valid_0's f1_gabri: 0.590019	valid_1's binary_logloss: 0.343953	valid_1's f1_gabri: 0.556423
[500]	valid_0's binary_logloss: 0.328789	valid_0's f1_gabri: 0.598273	valid_1's binary_logloss: 0.341266	valid_1's f1_gabri: 0.558291
[600]	valid_0's binary_logloss: 0.324369	valid_0's f1_gabri: 0.606209	valid_1's binary_logloss: 0.339302	valid_1's f1_gabri: 0.558175
[700]	valid_0's binary_logloss: 0.320354	valid_0's f1_gabri: 0.613862	valid_1's binary_logloss: 0.337633	valid_1's f1_gabri: 0.558797
[800]	valid_0's binary_logloss: 0.316569	valid_0's f1_gabri: 0.621561	valid_1's binary_logloss: 0.33603	valid_1's f1_gabri: 0.558841
[900]	valid_0's binary_logloss: 0.312931	valid_0's f1_gabri: 0.628802	valid_1's binary_logloss: 0.334541	valid_1's f1_gabri: 0.561382
[1000]	valid_0's binary_logloss: 0.309404	valid_0's f1_gabri: 0

[800]	valid_0's binary_logloss: 0.314913	valid_0's f1_gabri: 0.624426	valid_1's binary_logloss: 0.335008	valid_1's f1_gabri: 0.561231
[900]	valid_0's binary_logloss: 0.311117	valid_0's f1_gabri: 0.631573	valid_1's binary_logloss: 0.333497	valid_1's f1_gabri: 0.562884
[1000]	valid_0's binary_logloss: 0.307446	valid_0's f1_gabri: 0.638497	valid_1's binary_logloss: 0.331986	valid_1's f1_gabri: 0.563166
[1100]	valid_0's binary_logloss: 0.304024	valid_0's f1_gabri: 0.645307	valid_1's binary_logloss: 0.330602	valid_1's f1_gabri: 0.563127
[1200]	valid_0's binary_logloss: 0.300625	valid_0's f1_gabri: 0.651748	valid_1's binary_logloss: 0.329233	valid_1's f1_gabri: 0.564563
[1300]	valid_0's binary_logloss: 0.29731	valid_0's f1_gabri: 0.658294	valid_1's binary_logloss: 0.327926	valid_1's f1_gabri: 0.564106
Early stopping, best iteration is:
[1200]	valid_0's binary_logloss: 0.300625	valid_0's f1_gabri: 0.651748	valid_1's binary_logloss: 0.329233	valid_1's f1_gabri: 0.564563
0.5645625177121475
Trai

Training until validation scores don't improve for 150 rounds
[100]	valid_0's binary_logloss: 0.363677	valid_0's f1_gabri: 0.519844	valid_1's binary_logloss: 0.365968	valid_1's f1_gabri: 0.49555
[200]	valid_0's binary_logloss: 0.348056	valid_0's f1_gabri: 0.564248	valid_1's binary_logloss: 0.352912	valid_1's f1_gabri: 0.536733
[300]	valid_0's binary_logloss: 0.339211	valid_0's f1_gabri: 0.581102	valid_1's binary_logloss: 0.346662	valid_1's f1_gabri: 0.549923
[400]	valid_0's binary_logloss: 0.332677	valid_0's f1_gabri: 0.593537	valid_1's binary_logloss: 0.342582	valid_1's f1_gabri: 0.555222
[500]	valid_0's binary_logloss: 0.327442	valid_0's f1_gabri: 0.602193	valid_1's binary_logloss: 0.339865	valid_1's f1_gabri: 0.561535
[600]	valid_0's binary_logloss: 0.322809	valid_0's f1_gabri: 0.609872	valid_1's binary_logloss: 0.337647	valid_1's f1_gabri: 0.563677
[700]	valid_0's binary_logloss: 0.318634	valid_0's f1_gabri: 0.617362	valid_1's binary_logloss: 0.335827	valid_1's f1_gabri: 0.564419
[

[500]	valid_0's binary_logloss: 0.327625	valid_0's f1_gabri: 0.60042	valid_1's binary_logloss: 0.340394	valid_1's f1_gabri: 0.559537
[600]	valid_0's binary_logloss: 0.322994	valid_0's f1_gabri: 0.608676	valid_1's binary_logloss: 0.338305	valid_1's f1_gabri: 0.560858
[700]	valid_0's binary_logloss: 0.318697	valid_0's f1_gabri: 0.616193	valid_1's binary_logloss: 0.336459	valid_1's f1_gabri: 0.561356
[800]	valid_0's binary_logloss: 0.314682	valid_0's f1_gabri: 0.623791	valid_1's binary_logloss: 0.334718	valid_1's f1_gabri: 0.562149
[900]	valid_0's binary_logloss: 0.310955	valid_0's f1_gabri: 0.630652	valid_1's binary_logloss: 0.333204	valid_1's f1_gabri: 0.562778
[1000]	valid_0's binary_logloss: 0.307333	valid_0's f1_gabri: 0.636916	valid_1's binary_logloss: 0.331822	valid_1's f1_gabri: 0.562683
[1100]	valid_0's binary_logloss: 0.303699	valid_0's f1_gabri: 0.643981	valid_1's binary_logloss: 0.330406	valid_1's f1_gabri: 0.56108
Early stopping, best iteration is:
[982]	valid_0's binary_logl

[400]	valid_0's binary_logloss: 0.332533	valid_0's f1_gabri: 0.590543	valid_1's binary_logloss: 0.342715	valid_1's f1_gabri: 0.556324
[500]	valid_0's binary_logloss: 0.327126	valid_0's f1_gabri: 0.600345	valid_1's binary_logloss: 0.339891	valid_1's f1_gabri: 0.558083
[600]	valid_0's binary_logloss: 0.322468	valid_0's f1_gabri: 0.608737	valid_1's binary_logloss: 0.33773	valid_1's f1_gabri: 0.559043
[700]	valid_0's binary_logloss: 0.318159	valid_0's f1_gabri: 0.617449	valid_1's binary_logloss: 0.335874	valid_1's f1_gabri: 0.560676
[800]	valid_0's binary_logloss: 0.314214	valid_0's f1_gabri: 0.624635	valid_1's binary_logloss: 0.334193	valid_1's f1_gabri: 0.562205
[900]	valid_0's binary_logloss: 0.310341	valid_0's f1_gabri: 0.632322	valid_1's binary_logloss: 0.332653	valid_1's f1_gabri: 0.563017
[1000]	valid_0's binary_logloss: 0.306679	valid_0's f1_gabri: 0.639361	valid_1's binary_logloss: 0.331137	valid_1's f1_gabri: 0.563191
[1100]	valid_0's binary_logloss: 0.303071	valid_0's f1_gabri: 

[200]	valid_0's binary_logloss: 0.348244	valid_0's f1_gabri: 0.56257	valid_1's binary_logloss: 0.353041	valid_1's f1_gabri: 0.537734
[300]	valid_0's binary_logloss: 0.339411	valid_0's f1_gabri: 0.580259	valid_1's binary_logloss: 0.346904	valid_1's f1_gabri: 0.55319
[400]	valid_0's binary_logloss: 0.332894	valid_0's f1_gabri: 0.590401	valid_1's binary_logloss: 0.342989	valid_1's f1_gabri: 0.559901
[500]	valid_0's binary_logloss: 0.32749	valid_0's f1_gabri: 0.59855	valid_1's binary_logloss: 0.340059	valid_1's f1_gabri: 0.561992
[600]	valid_0's binary_logloss: 0.322785	valid_0's f1_gabri: 0.607019	valid_1's binary_logloss: 0.33778	valid_1's f1_gabri: 0.565335
[700]	valid_0's binary_logloss: 0.31849	valid_0's f1_gabri: 0.614802	valid_1's binary_logloss: 0.335872	valid_1's f1_gabri: 0.563542
Early stopping, best iteration is:
[610]	valid_0's binary_logloss: 0.322335	valid_0's f1_gabri: 0.608172	valid_1's binary_logloss: 0.337542	valid_1's f1_gabri: 0.565887
0.565887238623654
Training until 

Training until validation scores don't improve for 150 rounds
[100]	valid_0's binary_logloss: 0.364105	valid_0's f1_gabri: 0.516101	valid_1's binary_logloss: 0.366387	valid_1's f1_gabri: 0.494548
[200]	valid_0's binary_logloss: 0.348515	valid_0's f1_gabri: 0.562626	valid_1's binary_logloss: 0.353317	valid_1's f1_gabri: 0.538914
[300]	valid_0's binary_logloss: 0.339702	valid_0's f1_gabri: 0.579032	valid_1's binary_logloss: 0.347042	valid_1's f1_gabri: 0.550166
[400]	valid_0's binary_logloss: 0.333435	valid_0's f1_gabri: 0.591738	valid_1's binary_logloss: 0.343313	valid_1's f1_gabri: 0.554837
[500]	valid_0's binary_logloss: 0.328216	valid_0's f1_gabri: 0.600787	valid_1's binary_logloss: 0.340667	valid_1's f1_gabri: 0.558873
[600]	valid_0's binary_logloss: 0.323599	valid_0's f1_gabri: 0.608175	valid_1's binary_logloss: 0.338592	valid_1's f1_gabri: 0.560137
[700]	valid_0's binary_logloss: 0.319486	valid_0's f1_gabri: 0.616787	valid_1's binary_logloss: 0.33684	valid_1's f1_gabri: 0.560162
E

[100]	valid_0's binary_logloss: 0.363415	valid_0's f1_gabri: 0.518493	valid_1's binary_logloss: 0.365812	valid_1's f1_gabri: 0.493983
[200]	valid_0's binary_logloss: 0.347787	valid_0's f1_gabri: 0.565115	valid_1's binary_logloss: 0.352628	valid_1's f1_gabri: 0.538314
[300]	valid_0's binary_logloss: 0.339113	valid_0's f1_gabri: 0.580206	valid_1's binary_logloss: 0.346515	valid_1's f1_gabri: 0.549449
[400]	valid_0's binary_logloss: 0.332784	valid_0's f1_gabri: 0.591811	valid_1's binary_logloss: 0.342711	valid_1's f1_gabri: 0.555336
[500]	valid_0's binary_logloss: 0.327555	valid_0's f1_gabri: 0.600316	valid_1's binary_logloss: 0.340066	valid_1's f1_gabri: 0.559098
[600]	valid_0's binary_logloss: 0.322936	valid_0's f1_gabri: 0.607784	valid_1's binary_logloss: 0.338013	valid_1's f1_gabri: 0.56051
[700]	valid_0's binary_logloss: 0.318699	valid_0's f1_gabri: 0.614836	valid_1's binary_logloss: 0.336188	valid_1's f1_gabri: 0.561245
Early stopping, best iteration is:
[630]	valid_0's binary_loglo

[700]	valid_0's binary_logloss: 0.318779	valid_0's f1_gabri: 0.616534	valid_1's binary_logloss: 0.336285	valid_1's f1_gabri: 0.562424
[800]	valid_0's binary_logloss: 0.314767	valid_0's f1_gabri: 0.624629	valid_1's binary_logloss: 0.334608	valid_1's f1_gabri: 0.561103
Early stopping, best iteration is:
[707]	valid_0's binary_logloss: 0.318483	valid_0's f1_gabri: 0.617331	valid_1's binary_logloss: 0.336177	valid_1's f1_gabri: 0.562868
0.56286786139825
Training until validation scores don't improve for 150 rounds
[100]	valid_0's binary_logloss: 0.364302	valid_0's f1_gabri: 0.516668	valid_1's binary_logloss: 0.366514	valid_1's f1_gabri: 0.494786
[200]	valid_0's binary_logloss: 0.349076	valid_0's f1_gabri: 0.561976	valid_1's binary_logloss: 0.353874	valid_1's f1_gabri: 0.53377
[300]	valid_0's binary_logloss: 0.340265	valid_0's f1_gabri: 0.578853	valid_1's binary_logloss: 0.347675	valid_1's f1_gabri: 0.551421
[400]	valid_0's binary_logloss: 0.333726	valid_0's f1_gabri: 0.589614	valid_1's bin

Early stopping, best iteration is:
[565]	valid_0's binary_logloss: 0.324765	valid_0's f1_gabri: 0.605796	valid_1's binary_logloss: 0.338521	valid_1's f1_gabri: 0.564119
0.5641188099923393
Training until validation scores don't improve for 150 rounds
[100]	valid_0's binary_logloss: 0.363789	valid_0's f1_gabri: 0.517904	valid_1's binary_logloss: 0.366123	valid_1's f1_gabri: 0.494556
[200]	valid_0's binary_logloss: 0.34833	valid_0's f1_gabri: 0.563377	valid_1's binary_logloss: 0.353245	valid_1's f1_gabri: 0.53871
[300]	valid_0's binary_logloss: 0.339506	valid_0's f1_gabri: 0.580061	valid_1's binary_logloss: 0.347015	valid_1's f1_gabri: 0.551141
[400]	valid_0's binary_logloss: 0.333006	valid_0's f1_gabri: 0.591782	valid_1's binary_logloss: 0.343048	valid_1's f1_gabri: 0.559477
[500]	valid_0's binary_logloss: 0.327703	valid_0's f1_gabri: 0.600051	valid_1's binary_logloss: 0.340248	valid_1's f1_gabri: 0.562091
[600]	valid_0's binary_logloss: 0.32307	valid_0's f1_gabri: 0.607905	valid_1's bin

Training until validation scores don't improve for 150 rounds
[100]	valid_0's binary_logloss: 0.363626	valid_0's f1_gabri: 0.519603	valid_1's binary_logloss: 0.365886	valid_1's f1_gabri: 0.493556
[200]	valid_0's binary_logloss: 0.347978	valid_0's f1_gabri: 0.564282	valid_1's binary_logloss: 0.352723	valid_1's f1_gabri: 0.53736
[300]	valid_0's binary_logloss: 0.339148	valid_0's f1_gabri: 0.580756	valid_1's binary_logloss: 0.346512	valid_1's f1_gabri: 0.552143
[400]	valid_0's binary_logloss: 0.332745	valid_0's f1_gabri: 0.591518	valid_1's binary_logloss: 0.342688	valid_1's f1_gabri: 0.557042
[500]	valid_0's binary_logloss: 0.32757	valid_0's f1_gabri: 0.600313	valid_1's binary_logloss: 0.340016	valid_1's f1_gabri: 0.559191
[600]	valid_0's binary_logloss: 0.322963	valid_0's f1_gabri: 0.608957	valid_1's binary_logloss: 0.337932	valid_1's f1_gabri: 0.562407
[700]	valid_0's binary_logloss: 0.318823	valid_0's f1_gabri: 0.616051	valid_1's binary_logloss: 0.336153	valid_1's f1_gabri: 0.562299
[8

[800]	valid_0's binary_logloss: 0.314149	valid_0's f1_gabri: 0.624773	valid_1's binary_logloss: 0.334071	valid_1's f1_gabri: 0.562915
[900]	valid_0's binary_logloss: 0.310286	valid_0's f1_gabri: 0.632127	valid_1's binary_logloss: 0.332497	valid_1's f1_gabri: 0.563099
[1000]	valid_0's binary_logloss: 0.306605	valid_0's f1_gabri: 0.639074	valid_1's binary_logloss: 0.330954	valid_1's f1_gabri: 0.565636
[1100]	valid_0's binary_logloss: 0.303057	valid_0's f1_gabri: 0.645559	valid_1's binary_logloss: 0.32954	valid_1's f1_gabri: 0.565784
[1200]	valid_0's binary_logloss: 0.299698	valid_0's f1_gabri: 0.653289	valid_1's binary_logloss: 0.32821	valid_1's f1_gabri: 0.564507
Early stopping, best iteration is:
[1120]	valid_0's binary_logloss: 0.302395	valid_0's f1_gabri: 0.646978	valid_1's binary_logloss: 0.32927	valid_1's f1_gabri: 0.566542
0.5665415780356374
Training until validation scores don't improve for 150 rounds
[100]	valid_0's binary_logloss: 0.364051	valid_0's f1_gabri: 0.518467	valid_1's

[400]	valid_0's binary_logloss: 0.332562	valid_0's f1_gabri: 0.592272	valid_1's binary_logloss: 0.342364	valid_1's f1_gabri: 0.558989
[500]	valid_0's binary_logloss: 0.327169	valid_0's f1_gabri: 0.600367	valid_1's binary_logloss: 0.339463	valid_1's f1_gabri: 0.56282
[600]	valid_0's binary_logloss: 0.322453	valid_0's f1_gabri: 0.607784	valid_1's binary_logloss: 0.337165	valid_1's f1_gabri: 0.564261
[700]	valid_0's binary_logloss: 0.318227	valid_0's f1_gabri: 0.615766	valid_1's binary_logloss: 0.335257	valid_1's f1_gabri: 0.564626
[800]	valid_0's binary_logloss: 0.314167	valid_0's f1_gabri: 0.623584	valid_1's binary_logloss: 0.333534	valid_1's f1_gabri: 0.567189
[900]	valid_0's binary_logloss: 0.310336	valid_0's f1_gabri: 0.630469	valid_1's binary_logloss: 0.331931	valid_1's f1_gabri: 0.564669
Early stopping, best iteration is:
[844]	valid_0's binary_logloss: 0.312476	valid_0's f1_gabri: 0.626051	valid_1's binary_logloss: 0.332811	valid_1's f1_gabri: 0.567995
0.567994852867715
Training u

[700]	valid_0's binary_logloss: 0.318562	valid_0's f1_gabri: 0.617108	valid_1's binary_logloss: 0.336138	valid_1's f1_gabri: 0.562003
[800]	valid_0's binary_logloss: 0.31463	valid_0's f1_gabri: 0.624429	valid_1's binary_logloss: 0.334484	valid_1's f1_gabri: 0.561494
Early stopping, best iteration is:
[667]	valid_0's binary_logloss: 0.319977	valid_0's f1_gabri: 0.614359	valid_1's binary_logloss: 0.336793	valid_1's f1_gabri: 0.56243
0.5624297216134264
Training until validation scores don't improve for 150 rounds
[100]	valid_0's binary_logloss: 0.364471	valid_0's f1_gabri: 0.515622	valid_1's binary_logloss: 0.366679	valid_1's f1_gabri: 0.494418
[200]	valid_0's binary_logloss: 0.349449	valid_0's f1_gabri: 0.559903	valid_1's binary_logloss: 0.354267	valid_1's f1_gabri: 0.536794
[300]	valid_0's binary_logloss: 0.340797	valid_0's f1_gabri: 0.577178	valid_1's binary_logloss: 0.348184	valid_1's f1_gabri: 0.548884
[400]	valid_0's binary_logloss: 0.334511	valid_0's f1_gabri: 0.587433	valid_1's bi

Training until validation scores don't improve for 150 rounds
[100]	valid_0's binary_logloss: 0.363458	valid_0's f1_gabri: 0.517686	valid_1's binary_logloss: 0.365706	valid_1's f1_gabri: 0.491313
[200]	valid_0's binary_logloss: 0.347637	valid_0's f1_gabri: 0.565995	valid_1's binary_logloss: 0.352477	valid_1's f1_gabri: 0.536088
[300]	valid_0's binary_logloss: 0.338613	valid_0's f1_gabri: 0.582977	valid_1's binary_logloss: 0.346147	valid_1's f1_gabri: 0.550625
[400]	valid_0's binary_logloss: 0.331965	valid_0's f1_gabri: 0.593332	valid_1's binary_logloss: 0.342052	valid_1's f1_gabri: 0.558655
[500]	valid_0's binary_logloss: 0.326587	valid_0's f1_gabri: 0.602789	valid_1's binary_logloss: 0.339247	valid_1's f1_gabri: 0.562241
[600]	valid_0's binary_logloss: 0.321953	valid_0's f1_gabri: 0.611056	valid_1's binary_logloss: 0.33713	valid_1's f1_gabri: 0.563619
[700]	valid_0's binary_logloss: 0.317732	valid_0's f1_gabri: 0.618999	valid_1's binary_logloss: 0.33529	valid_1's f1_gabri: 0.564506
[8

[500]	valid_0's binary_logloss: 0.327271	valid_0's f1_gabri: 0.600491	valid_1's binary_logloss: 0.339492	valid_1's f1_gabri: 0.560981
[600]	valid_0's binary_logloss: 0.322663	valid_0's f1_gabri: 0.608851	valid_1's binary_logloss: 0.337323	valid_1's f1_gabri: 0.561522
[700]	valid_0's binary_logloss: 0.318478	valid_0's f1_gabri: 0.616362	valid_1's binary_logloss: 0.335492	valid_1's f1_gabri: 0.563195
[800]	valid_0's binary_logloss: 0.31451	valid_0's f1_gabri: 0.624154	valid_1's binary_logloss: 0.333866	valid_1's f1_gabri: 0.564815
[900]	valid_0's binary_logloss: 0.310754	valid_0's f1_gabri: 0.631455	valid_1's binary_logloss: 0.3323	valid_1's f1_gabri: 0.563328
Early stopping, best iteration is:
[789]	valid_0's binary_logloss: 0.314914	valid_0's f1_gabri: 0.623619	valid_1's binary_logloss: 0.33401	valid_1's f1_gabri: 0.565199
0.5651989519721818
Training until validation scores don't improve for 150 rounds
[100]	valid_0's binary_logloss: 0.36389	valid_0's f1_gabri: 0.516767	valid_1's binar

[700]	valid_0's binary_logloss: 0.318001	valid_0's f1_gabri: 0.616911	valid_1's binary_logloss: 0.335582	valid_1's f1_gabri: 0.564315
[800]	valid_0's binary_logloss: 0.313969	valid_0's f1_gabri: 0.62549	valid_1's binary_logloss: 0.333809	valid_1's f1_gabri: 0.565389
[900]	valid_0's binary_logloss: 0.310201	valid_0's f1_gabri: 0.632571	valid_1's binary_logloss: 0.33227	valid_1's f1_gabri: 0.565452
[1000]	valid_0's binary_logloss: 0.306489	valid_0's f1_gabri: 0.639183	valid_1's binary_logloss: 0.330675	valid_1's f1_gabri: 0.565572
Early stopping, best iteration is:
[893]	valid_0's binary_logloss: 0.310453	valid_0's f1_gabri: 0.632204	valid_1's binary_logloss: 0.332377	valid_1's f1_gabri: 0.565742
0.5657420898034925
Training until validation scores don't improve for 150 rounds
[100]	valid_0's binary_logloss: 0.363455	valid_0's f1_gabri: 0.519766	valid_1's binary_logloss: 0.365762	valid_1's f1_gabri: 0.493343
[200]	valid_0's binary_logloss: 0.347739	valid_0's f1_gabri: 0.564191	valid_1's b

[1000]	valid_0's binary_logloss: 0.306333	valid_0's f1_gabri: 0.640768	valid_1's binary_logloss: 0.330561	valid_1's f1_gabri: 0.568166
[1100]	valid_0's binary_logloss: 0.302764	valid_0's f1_gabri: 0.647719	valid_1's binary_logloss: 0.329098	valid_1's f1_gabri: 0.567743
[1200]	valid_0's binary_logloss: 0.299402	valid_0's f1_gabri: 0.654628	valid_1's binary_logloss: 0.327779	valid_1's f1_gabri: 0.567451
Early stopping, best iteration is:
[1062]	valid_0's binary_logloss: 0.30405	valid_0's f1_gabri: 0.645123	valid_1's binary_logloss: 0.329632	valid_1's f1_gabri: 0.56958
0.5695803492031657
Training until validation scores don't improve for 150 rounds
[100]	valid_0's binary_logloss: 0.363762	valid_0's f1_gabri: 0.519779	valid_1's binary_logloss: 0.366109	valid_1's f1_gabri: 0.493538
[200]	valid_0's binary_logloss: 0.347973	valid_0's f1_gabri: 0.563593	valid_1's binary_logloss: 0.352722	valid_1's f1_gabri: 0.538328
[300]	valid_0's binary_logloss: 0.338863	valid_0's f1_gabri: 0.583376	valid_1'

Training until validation scores don't improve for 150 rounds
[100]	valid_0's binary_logloss: 0.363707	valid_0's f1_gabri: 0.518817	valid_1's binary_logloss: 0.366007	valid_1's f1_gabri: 0.495771
[200]	valid_0's binary_logloss: 0.347987	valid_0's f1_gabri: 0.564265	valid_1's binary_logloss: 0.35273	valid_1's f1_gabri: 0.537357
[300]	valid_0's binary_logloss: 0.338915	valid_0's f1_gabri: 0.582577	valid_1's binary_logloss: 0.34629	valid_1's f1_gabri: 0.552394
[400]	valid_0's binary_logloss: 0.332313	valid_0's f1_gabri: 0.593516	valid_1's binary_logloss: 0.342257	valid_1's f1_gabri: 0.557049
[500]	valid_0's binary_logloss: 0.326855	valid_0's f1_gabri: 0.602448	valid_1's binary_logloss: 0.339391	valid_1's f1_gabri: 0.561342
[600]	valid_0's binary_logloss: 0.322199	valid_0's f1_gabri: 0.609676	valid_1's binary_logloss: 0.337199	valid_1's f1_gabri: 0.563873
[700]	valid_0's binary_logloss: 0.317835	valid_0's f1_gabri: 0.618524	valid_1's binary_logloss: 0.335269	valid_1's f1_gabri: 0.564042
[8

[200]	valid_0's binary_logloss: 0.348036	valid_0's f1_gabri: 0.563187	valid_1's binary_logloss: 0.352865	valid_1's f1_gabri: 0.535337
[300]	valid_0's binary_logloss: 0.338923	valid_0's f1_gabri: 0.582657	valid_1's binary_logloss: 0.346457	valid_1's f1_gabri: 0.549924
[400]	valid_0's binary_logloss: 0.332331	valid_0's f1_gabri: 0.59315	valid_1's binary_logloss: 0.342346	valid_1's f1_gabri: 0.556833
[500]	valid_0's binary_logloss: 0.326936	valid_0's f1_gabri: 0.601789	valid_1's binary_logloss: 0.339391	valid_1's f1_gabri: 0.562367
[600]	valid_0's binary_logloss: 0.322315	valid_0's f1_gabri: 0.609196	valid_1's binary_logloss: 0.33719	valid_1's f1_gabri: 0.563469
[700]	valid_0's binary_logloss: 0.317977	valid_0's f1_gabri: 0.616626	valid_1's binary_logloss: 0.335243	valid_1's f1_gabri: 0.563747
[800]	valid_0's binary_logloss: 0.313901	valid_0's f1_gabri: 0.624535	valid_1's binary_logloss: 0.333542	valid_1's f1_gabri: 0.563105
Early stopping, best iteration is:
[716]	valid_0's binary_loglos

[600]	valid_0's binary_logloss: 0.322037	valid_0's f1_gabri: 0.610477	valid_1's binary_logloss: 0.337021	valid_1's f1_gabri: 0.564847
[700]	valid_0's binary_logloss: 0.317711	valid_0's f1_gabri: 0.618461	valid_1's binary_logloss: 0.335146	valid_1's f1_gabri: 0.565911
[800]	valid_0's binary_logloss: 0.313685	valid_0's f1_gabri: 0.626389	valid_1's binary_logloss: 0.333481	valid_1's f1_gabri: 0.565586
Early stopping, best iteration is:
[689]	valid_0's binary_logloss: 0.318168	valid_0's f1_gabri: 0.617528	valid_1's binary_logloss: 0.33531	valid_1's f1_gabri: 0.566592
0.5665918992960548
Training until validation scores don't improve for 150 rounds
[100]	valid_0's binary_logloss: 0.363292	valid_0's f1_gabri: 0.519357	valid_1's binary_logloss: 0.365552	valid_1's f1_gabri: 0.492103
[200]	valid_0's binary_logloss: 0.347649	valid_0's f1_gabri: 0.565786	valid_1's binary_logloss: 0.352437	valid_1's f1_gabri: 0.537554
[300]	valid_0's binary_logloss: 0.338524	valid_0's f1_gabri: 0.582601	valid_1's b

[500]	valid_0's binary_logloss: 0.327491	valid_0's f1_gabri: 0.601339	valid_1's binary_logloss: 0.339881	valid_1's f1_gabri: 0.559837
[600]	valid_0's binary_logloss: 0.32279	valid_0's f1_gabri: 0.609531	valid_1's binary_logloss: 0.337667	valid_1's f1_gabri: 0.560488
[700]	valid_0's binary_logloss: 0.318543	valid_0's f1_gabri: 0.616757	valid_1's binary_logloss: 0.335737	valid_1's f1_gabri: 0.56137
[800]	valid_0's binary_logloss: 0.31453	valid_0's f1_gabri: 0.624173	valid_1's binary_logloss: 0.33407	valid_1's f1_gabri: 0.562188
[900]	valid_0's binary_logloss: 0.31078	valid_0's f1_gabri: 0.63102	valid_1's binary_logloss: 0.33253	valid_1's f1_gabri: 0.563999
[1000]	valid_0's binary_logloss: 0.3072	valid_0's f1_gabri: 0.638605	valid_1's binary_logloss: 0.331057	valid_1's f1_gabri: 0.563714
[1100]	valid_0's binary_logloss: 0.303675	valid_0's f1_gabri: 0.645748	valid_1's binary_logloss: 0.329586	valid_1's f1_gabri: 0.565542
[1200]	valid_0's binary_logloss: 0.300199	valid_0's f1_gabri: 0.65187

0.568463687661077
IT  8 best cost  0.5700718576046943
['CODE_GENDER', 'FLAG_OWN_CAR', 'FLAG_OWN_REALTY', 'AMT_INCOME_TOTAL', 'AMT_ANNUITY', 'AMT_GOODS_PRICE', 'DAYS_BIRTH', 'DAYS_ID_PUBLISH', 'FLAG_WORK_PHONE', 'FLAG_CONT_MOBILE', 'FLAG_EMAIL', 'CNT_FAM_MEMBERS', 'REGION_RATING_CLIENT', 'REG_REGION_NOT_LIVE_REGION', 'REG_REGION_NOT_WORK_REGION', 'LIVE_REGION_NOT_WORK_REGION', 'REG_CITY_NOT_LIVE_CITY', 'REG_CITY_NOT_WORK_CITY', 'LIVE_CITY_NOT_WORK_CITY', 'EXT_SOURCE_1', 'EXT_SOURCE_2', 'EXT_SOURCE_3', 'APARTMENTS_AVG', 'YEARS_BEGINEXPLUATATION_AVG', 'COMMONAREA_AVG', 'ELEVATORS_AVG', 'FLOORSMAX_AVG', 'FLOORSMIN_AVG', 'APARTMENTS_MODE', 'YEARS_BEGINEXPLUATATION_MODE', 'YEARS_BUILD_MODE', 'ELEVATORS_MODE', 'FLOORSMAX_MODE', 'FLOORSMIN_MODE', 'LANDAREA_MODE', 'LIVINGAREA_MODE', 'APARTMENTS_MEDI', 'YEARS_BEGINEXPLUATATION_MEDI', 'ENTRANCES_MEDI', 'FLOORSMAX_MEDI', 'FLOORSMIN_MEDI', 'LANDAREA_MEDI', 'LIVINGAREA_MEDI', 'NONLIVINGAPARTMENTS_MEDI', 'TOTALAREA_MODE', 'OBS_60_CNT_SOCIAL_CIRCLE', 

Training until validation scores don't improve for 150 rounds
[100]	valid_0's binary_logloss: 0.363671	valid_0's f1_gabri: 0.519197	valid_1's binary_logloss: 0.366084	valid_1's f1_gabri: 0.494125
[200]	valid_0's binary_logloss: 0.347985	valid_0's f1_gabri: 0.564328	valid_1's binary_logloss: 0.352986	valid_1's f1_gabri: 0.533253
[300]	valid_0's binary_logloss: 0.338922	valid_0's f1_gabri: 0.583332	valid_1's binary_logloss: 0.346524	valid_1's f1_gabri: 0.552774
[400]	valid_0's binary_logloss: 0.332399	valid_0's f1_gabri: 0.594079	valid_1's binary_logloss: 0.342531	valid_1's f1_gabri: 0.558167
[500]	valid_0's binary_logloss: 0.327009	valid_0's f1_gabri: 0.601799	valid_1's binary_logloss: 0.33972	valid_1's f1_gabri: 0.560607
[600]	valid_0's binary_logloss: 0.322328	valid_0's f1_gabri: 0.610297	valid_1's binary_logloss: 0.337609	valid_1's f1_gabri: 0.561605
[700]	valid_0's binary_logloss: 0.318005	valid_0's f1_gabri: 0.618613	valid_1's binary_logloss: 0.335754	valid_1's f1_gabri: 0.563282
[

[1100]	valid_0's binary_logloss: 0.303863	valid_0's f1_gabri: 0.64499	valid_1's binary_logloss: 0.330102	valid_1's f1_gabri: 0.567375
Early stopping, best iteration is:
[960]	valid_0's binary_logloss: 0.308808	valid_0's f1_gabri: 0.634693	valid_1's binary_logloss: 0.332123	valid_1's f1_gabri: 0.569493
0.5694926063889668
Training until validation scores don't improve for 150 rounds
[100]	valid_0's binary_logloss: 0.363611	valid_0's f1_gabri: 0.518465	valid_1's binary_logloss: 0.365947	valid_1's f1_gabri: 0.494933
[200]	valid_0's binary_logloss: 0.347966	valid_0's f1_gabri: 0.563489	valid_1's binary_logloss: 0.352752	valid_1's f1_gabri: 0.537949
[300]	valid_0's binary_logloss: 0.338888	valid_0's f1_gabri: 0.581669	valid_1's binary_logloss: 0.346386	valid_1's f1_gabri: 0.551199
[400]	valid_0's binary_logloss: 0.332283	valid_0's f1_gabri: 0.592497	valid_1's binary_logloss: 0.342287	valid_1's f1_gabri: 0.560694
[500]	valid_0's binary_logloss: 0.326882	valid_0's f1_gabri: 0.60141	valid_1's b

[700]	valid_0's binary_logloss: 0.318594	valid_0's f1_gabri: 0.616092	valid_1's binary_logloss: 0.335778	valid_1's f1_gabri: 0.565363
[800]	valid_0's binary_logloss: 0.314647	valid_0's f1_gabri: 0.623259	valid_1's binary_logloss: 0.334111	valid_1's f1_gabri: 0.565131
[900]	valid_0's binary_logloss: 0.310945	valid_0's f1_gabri: 0.629954	valid_1's binary_logloss: 0.332642	valid_1's f1_gabri: 0.566762
[1000]	valid_0's binary_logloss: 0.307305	valid_0's f1_gabri: 0.637002	valid_1's binary_logloss: 0.331211	valid_1's f1_gabri: 0.565589
Early stopping, best iteration is:
[892]	valid_0's binary_logloss: 0.311255	valid_0's f1_gabri: 0.62981	valid_1's binary_logloss: 0.332759	valid_1's f1_gabri: 0.567178
0.5671782088520807
Training until validation scores don't improve for 150 rounds
[100]	valid_0's binary_logloss: 0.363805	valid_0's f1_gabri: 0.517339	valid_1's binary_logloss: 0.366079	valid_1's f1_gabri: 0.49606
[200]	valid_0's binary_logloss: 0.34831	valid_0's f1_gabri: 0.563115	valid_1's bi

[1100]	valid_0's binary_logloss: 0.302654	valid_0's f1_gabri: 0.646661	valid_1's binary_logloss: 0.328807	valid_1's f1_gabri: 0.569389
[1200]	valid_0's binary_logloss: 0.299251	valid_0's f1_gabri: 0.653625	valid_1's binary_logloss: 0.327438	valid_1's f1_gabri: 0.569454
[1300]	valid_0's binary_logloss: 0.295823	valid_0's f1_gabri: 0.660787	valid_1's binary_logloss: 0.326114	valid_1's f1_gabri: 0.568342
Early stopping, best iteration is:
[1205]	valid_0's binary_logloss: 0.299071	valid_0's f1_gabri: 0.653904	valid_1's binary_logloss: 0.32736	valid_1's f1_gabri: 0.569704
0.5697035652780225
Training until validation scores don't improve for 150 rounds
[100]	valid_0's binary_logloss: 0.363701	valid_0's f1_gabri: 0.517709	valid_1's binary_logloss: 0.365991	valid_1's f1_gabri: 0.492789
[200]	valid_0's binary_logloss: 0.34812	valid_0's f1_gabri: 0.562591	valid_1's binary_logloss: 0.35291	valid_1's f1_gabri: 0.537923
[300]	valid_0's binary_logloss: 0.339136	valid_0's f1_gabri: 0.581454	valid_1's

In [28]:
bestcost

0.5700957342739739

In [29]:
solution

['CODE_GENDER',
 'AMT_INCOME_TOTAL',
 'AMT_ANNUITY',
 'AMT_GOODS_PRICE',
 'DAYS_BIRTH',
 'DAYS_ID_PUBLISH',
 'FLAG_WORK_PHONE',
 'FLAG_CONT_MOBILE',
 'FLAG_EMAIL',
 'REGION_RATING_CLIENT',
 'REG_REGION_NOT_LIVE_REGION',
 'REG_REGION_NOT_WORK_REGION',
 'LIVE_REGION_NOT_WORK_REGION',
 'REG_CITY_NOT_LIVE_CITY',
 'REG_CITY_NOT_WORK_CITY',
 'LIVE_CITY_NOT_WORK_CITY',
 'EXT_SOURCE_1',
 'EXT_SOURCE_2',
 'EXT_SOURCE_3',
 'BASEMENTAREA_AVG',
 'COMMONAREA_AVG',
 'ELEVATORS_AVG',
 'ENTRANCES_AVG',
 'FLOORSMAX_AVG',
 'FLOORSMIN_AVG',
 'LIVINGAPARTMENTS_AVG',
 'LIVINGAREA_AVG',
 'APARTMENTS_MODE',
 'YEARS_BEGINEXPLUATATION_MODE',
 'YEARS_BUILD_MODE',
 'ELEVATORS_MODE',
 'FLOORSMAX_MODE',
 'LANDAREA_MODE',
 'LIVINGAREA_MODE',
 'NONLIVINGAREA_MODE',
 'APARTMENTS_MEDI',
 'YEARS_BEGINEXPLUATATION_MEDI',
 'FLOORSMAX_MEDI',
 'FLOORSMIN_MEDI',
 'LIVINGAREA_MEDI',
 'NONLIVINGAPARTMENTS_MEDI',
 'TOTALAREA_MODE',
 'OBS_30_CNT_SOCIAL_CIRCLE',
 'OBS_60_CNT_SOCIAL_CIRCLE',
 'DEF_60_CNT_SOCIAL_CIRCLE',
 'FLAG_DO

In [30]:
history

[0.561557156261801,
 0.5603901243323224,
 0.5639236163865882,
 0.5695819717921313,
 0.5688336972899707,
 0.5688496086370187,
 0.5700957342739739,
 0.5695803492031657,
 0.5700718576046943,
 0.5699602891159371]

El algoritmo evolutivo a logrado un f1 score geométrico de 0.57 con 410 variables.

In [31]:
len(solution)

410

In [38]:
datos_edaspy = datos.loc[:, solution]
datos_edaspy

CODE_GENDER  AMT_INCOME_TOTAL  AMT_ANNUITY  AMT_GOODS_PRICE  \
0                 0          202500.0      24700.5         351000.0   
1                 1          270000.0      35698.5        1129500.0   
2                 0           67500.0       6750.0         135000.0   
3                 1          135000.0      29686.5         297000.0   
4                 0          121500.0      21865.5         513000.0   
...             ...               ...          ...              ...   
307502            0          157500.0      27558.0         225000.0   
307503            1           72000.0      12001.5         225000.0   
307504            1          153000.0      29979.0         585000.0   
307505            1          171000.0      20205.0         319500.0   
307506            1          157500.0      49117.5         675000.0   

        DAYS_BIRTH  DAYS_ID_PUBLISH  FLAG_WORK_PHONE  FLAG_CONT_MOBILE  \
0            -9461            -2120                0                 1   
1           -16765             -291                0                 1   
2           -19046            -2531                1                 1   
3           -19005            -2437                0                 1   
4           -19932            -3458                0                 1   
...            ...              ...              ...               ...   
307502       -9327            -1982                0                 1   
307503      -20775            -4090                0                 1   
307504      -14966            -5150                0                 1   
307505      -11961             -931                0                 1   
307506      -16856             -410                1                 1   

        FLAG_EMAIL  REGION_RATING_CLIENT  ...  \
0                0                     2  ...   
1                0                     1  ...   
2                0                     2  ...   
3                0                     2  ...   
4                0                     2  ...   
...            ...                   ...  ...   
307502           0                     1  ...   
307503           0                     2  ...   
307504           1                     3  ...   
307505           0                     2  ...   
307506           0                     1  ...   

        CC_NAME_CONTRACT_STATUS_Refused_SUM  \
0                                       NaN   
1                                       NaN   
2                                       NaN   
3                                       0.0   
4                                       NaN   
...                                     ...   
307502                                  NaN   
307503                                  NaN   
307504                                  NaN   
307505                                  NaN   
307506                                  NaN   

        CC_NAME_CONTRACT_STATUS_Sent proposal_MIN  \
0                                             NaN   
1                                             NaN   
2                                             NaN   
3                                             0.0   
4                                             NaN   
...                                           ...   
307502                                        NaN   
307503                                        NaN   
307504                                        NaN   
307505                                        NaN   
307506                                        NaN   

        CC_NAME_CONTRACT_STATUS_Sent proposal_MEAN  \
0                                              NaN   
1                                              NaN   
2                                              NaN   
3                                              0.0   
4                                              NaN   
...                                            ...   
307502                                         NaN   
307503                                     

In [34]:
datos_edaspy.to_csv("../data/datos_edaspy.csv", index = False)